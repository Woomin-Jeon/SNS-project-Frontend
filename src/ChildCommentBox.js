/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { addChildComment } from './function';

function ChildCommentBox({
  commentState,
  setCommentState,
  parentsComment,
  currentUserState,
}) {
  const { id, userName } = currentUserState;
  const [temptState, setTemptState] = useState('');

  const handleAddChildComment = () => {
    if (temptState.trim()) {
      setCommentState(addChildComment(commentState, temptState, id, userName, parentsComment));
      setTemptState('');
    }
  };

  return (
    <div className="child-comment">
      └ {userName} : <input type="text" value={temptState} onChange={(e) => setTemptState(e.target.value)} />
      <button type="button" className="comment-input" onClick={handleAddChildComment}>입력</button>
    </div>
  );
}

export default ChildCommentBox;
