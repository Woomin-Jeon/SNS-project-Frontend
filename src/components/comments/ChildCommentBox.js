import React, { useState } from 'react';
import { addChildComment } from '../../function';

function ChildCommentBox({
  commentState,
  setCommentState,
  parentsComment,
  currentUserState,
}) {
  const { id, userName } = currentUserState;
  const { uniqueKey } = parentsComment;
  const [temptState, setTemptState] = useState('');

  const handleAddChildComment = async () => {
    if (temptState.trim()) {
      const { postComments } = await addChildComment(uniqueKey , temptState, id, userName);

      setCommentState({ ...commentState, comment : [...postComments.comment]});
    }
  };

  return (
    <div className="child-comment">
      └ {userName} : <input type="text" value={temptState} onChange={(e) => setTemptState(e.target.value)} />
      <button type="button" className="comment-input" onClick={handleAddChildComment}>입력</button>
    </div>
  );
}

export default ChildCommentBox;
