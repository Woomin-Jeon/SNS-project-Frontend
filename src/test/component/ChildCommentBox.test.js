import React from 'react';
import {shallow} from "enzyme";
import ChildCommentBox from "../../components/comments/ChildCommentBox";

describe('ChildCommentBox', () => {
  let commentState;
  let setCommentState;
  let currentUserState;
  let p;

  beforeEach(() => {
    currentUserState = {
      id: 'Random_ID',
      userName: 'Woomin'
    };
    p = {
      uniqueKey: 999,
    }
  });

  it('renders well', () => {
    const component = shallow(
      <ChildCommentBox
        commentState={commentState}
        setCommentState={setCommentState}
        parentsComment={p}
        currentUserState={currentUserState}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toMatch('Woomin');
    expect(component.find('button').text()).toBe('입력');
  });
});
