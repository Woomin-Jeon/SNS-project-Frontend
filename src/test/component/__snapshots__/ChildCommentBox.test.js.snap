// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildCommentBox renders well 1`] = `ShallowWrapper {}`;
